_posts/2019-02-07-partner-api-status-endpoint.md

```diff
@@ -0,0 +1,138 @@
+---
+layout: post
+title:  "Partner API Status Endpoint"
+date:   2019-02-07 4:00:00 -0400
+categories: api
+author: Kyle Chan
+---
+
+*Now that you have submitted your first deal via our [Lead Submission Endpoint](http://tech.forwardfinancing.com/api/2017/03/07/using-our-lead-submissions-api.html),
+you're probably wondering how it's doing.*
+
+## Getting Started
+The only things you'll need are: 
+- The lead `id` you received when you first submitted your deal
+- Your `api_key`
+
+### Authentication
+
+You will need to contact us to get your `api_key`. The API key should be included
+in your request headers such that the request header key is `api_key` and the
+request header value for that key is your `api_key`. If you omit the api_key or
+provide an incorrect one, the HTTP status code of the response will be 401
+Unauthorized.
+
+### HTTPS
+
+Since your API requests will contain sensitive personal information about our
+customers, you must use SSL (HTTPS). To discourage insecure API requests,
+the response code will be 403 Forbidden for all requests made over HTTP.
+
+### Rate Limiting
+
+We impose a rate limit per user of 1 request per second. If you exceed this rate
+of requests, the response code will be 429 Too Many Requests. If you need to
+exceed this limit for some reason, please let us know!
+
+## The Deal Status Endpoint
+
+You will be able to check your deal status with a GET request to the `/v1/deal_status/LEAD_ID` 
+endpoint with the `id` of your lead.
+For example, the GET url might be
+
+`https://api-staging.forwardfinancing.com/v1/deal_status/<LEAD_ID>`
```

Comment:
If we want to provide them staging to test with that's fine, but we should probably make it clear this is for testing/development, and that the real endpoint will be `https://api.forwardfinancing.com/v1/deal_status/<LEAD_ID>`.

Reply:
As to why we're not putting the real endpoint here: "We don't want partners to be sending us random applications while they're figuring out our API" - @zcotter

Reply:
@zcotter how do we communicate the actual endpoints then? I mean, people could probably guess it here, but it's something I's expect in the documentation of an API.

Reply:
Partners get a initial setup email with their API key that lists the staging and prod endpoints

Reply:
OK fair enough. I don't love it, but w/e.